src/Interpreters/InterpreterSelectQuery.h

```diff
@@ -205,6 +208,7 @@ class InterpreterSelectQuery : public IInterpreterUnionOrSelectQuery
     StorageSnapshotPtr storage_snapshot;
 
     /// Reuse already built sets for multiple passes of analysis, possibly across interpreters.
+    std::shared_ptr<SubqueriesForSets> subquery_for_sets;
```

Comment:
The reason to use `std::shared_ptr` is somehow hacky: to make it copyable.

Reply:
I understand but I didn't see where it's used for the creation of the copy.

Reply:
It's because we need `SelectQueryInfo` to be copyable.  

https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/StorageMerge.cpp#L218

I'll use a different implementation.